File: Top 100 Codes/06-Advanced Codes related to Arrays/28-Median of 2 sorted arrays of equal size/Medianof2sortedarraysofequalsize.cpp
#include <iostream>
#include <vector>
using namespace std;

float findMedianSortedArrays(vector<int>& a, vector<int>& b) {
    int n = a.size();
    vector<int> merged(2 * n);
    int i = 0, j = 0, k = 0;

    // Merge the two arrays
    while (i < n && j < n) {
        if (a[i] < b[j]) {
            merged[k++] = a[i++];
        } else {
            merged[k++] = b[j++];
        }
    }
    while (i < n) merged[k++] = a[i++];
    while (j < n) merged[k++] = b[j++];

    // Median of even size = average of middle two
    return (merged[n - 1] + merged[n]) / 2.0;
}

int main() {
    vector<int> a = {1, 3, 5};
    vector<int> b = {2, 4, 6};
    float median = findMedianSortedArrays(a, b);
    cout << "Median is: " << median << endl;
    return 0;
}

File: Top 100 Codes/06-Advanced Codes related to Arrays/28-Median of 2 sorted arrays of equal size/Medianof2sortedarraysofequalsize.py
def find_median_sorted_arrays(a, b):
    n = len(a)
    merged = []
    i = j = 0

    # Merge the arrays
    while i < n and j < n:
        if a[i] < b[j]:
            merged.append(a[i])
            i += 1
        else:
            merged.append(b[j])
            j += 1

    while i < n:
        merged.append(a[i])
        i += 1
    while j < n:
        merged.append(b[j])
        j += 1

    # Median for even total length
    return (merged[n - 1] + merged[n]) / 2

# Example usage
a = [1, 3, 5]
b = [2, 4, 6]
median = find_median_sorted_arrays(a, b)
print("Median is:", median)
